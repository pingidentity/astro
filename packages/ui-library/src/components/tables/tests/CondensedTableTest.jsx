jest.dontMock("../CondensedTable");

describe("CondensedTable", function () {
    var React = require("react"),
        ReactTestUtils = require("react-dom/test-utils"),
        TestUtils = require("../../../testutil/TestUtils"),
        CondensedTable = require("../CondensedTable");

    var mockData = {
        head: [
            "name",
            "age",
            "city"
        ],
        body: [
            [
                "tom",
                "25",
                "denver"
            ],
            [
                "jane",
                "36",
                "breckenridge"
            ],
            [
                "roy",
                "19",
                "arvada"
            ]
        ]
    };
    var dataObjects = [
        {
            name: "tom",
            age: 25,
            city: "denver"
        },
        {
            name: "jane",
            age: 36,
            city: "breckenridge"
        },
        {
            name: "roy",
            age: 19,
            city: "arvada"
        }
    ];

    it("data-id's don't change", () => {
        TestUtils.mountSnapshotDataIds(
            <CondensedTable
                headData={mockData.head}
                bodyData={mockData.body}
            />
        );
    });

    it("data-id's don't change with data object", () => {
        TestUtils.mountSnapshotDataIds(
            <CondensedTable
                data={dataObjects}
            />
        );
    });

    it("renders table with head and body", function () {
        var component = ReactTestUtils.renderIntoDocument(
            <div><CondensedTable headData={mockData.head} bodyData={mockData.body} /></div>);

        var section = TestUtils.findRenderedDOMNodeWithDataId(component, "section_0");
        var items = TestUtils.scryRenderedDOMNodesWithClass(section, "data-item");
        var label = TestUtils.findRenderedDOMNodeWithTag(items[0], "label");
        var description = TestUtils.findRenderedDOMNodeWithTag(items[0], "span");

        expect(items.length).toBe(3);
        expect(label.textContent).toBe("name");
        expect(description.textContent).toBe("tom");

    });

    it("renders table with data objects", function () {
        var component = ReactTestUtils.renderIntoDocument(<div><CondensedTable data={dataObjects} /></div>);
        var section = TestUtils.findRenderedDOMNodeWithDataId(component, "section_1");
        var items = TestUtils.scryRenderedDOMNodesWithClass(section, "data-item");
        var label = TestUtils.findRenderedDOMNodeWithTag(items[1], "label");
        var description = TestUtils.findRenderedDOMNodeWithTag(items[1], "span");

        expect(items.length).toBe(3);
        expect(label.textContent).toBe("age");
        expect(description.textContent).toBe("36");
    });
});
