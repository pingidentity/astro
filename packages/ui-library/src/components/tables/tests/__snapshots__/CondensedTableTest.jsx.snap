// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CondensedTable data-id's don't change 1`] = `
Array [
  "condensedTable",
  "section_0",
  "item_tom",
  "item_25",
  "item_denver",
  "section_1",
  "item_jane",
  "item_36",
  "item_breckenridge",
  "section_2",
  "item_roy",
  "item_19",
  "item_arvada",
]
`;

exports[`CondensedTable data-id's don't change with data object 1`] = `
Array [
  "condensedTable",
  "section_0",
  "item_tom",
  "item_25",
  "item_denver",
  "section_1",
  "item_jane",
  "item_36",
  "item_breckenridge",
  "section_2",
  "item_roy",
  "item_19",
  "item_arvada",
]
`;
